### ISSUE TITLE
Time series lags and leads

### ISSUE THREAD

**Issue by jaromir-benes** [IRIS-Solutions-Team/irispie#3]
### Lag/lead syntax modified

The syntax has been modified and uses square brackets now, e.g.

```
x[-1]
```

You can chain the lags/leads and data retrieval in any possible way

```
x[-1][from_date>>to_date]
x[-1](from_date>>to_date)
x[-1][from_date>>to_date][+1]
```
etc.

### Rationale behind square brackets

It is nice to have consistency between lags/leads syntax in model files (lags/leads of model variables) and in data processing scripts (lags/leads of time series).

In Python, only square and round brackets are available for indexing (no curly braces). Therefore, we use square brackets for lags/leads of time series, and introduce an optional syntax based on square brackets for lags/leads of model variables in model files, in parallel with the existing square brackets.
